### ISSUE TITLE
Why is the Loss (L^x + L^z) so small?

### ISSUE THREAD

**Issue by lovecambi** [ericjang/draw#2]
It is about 70, lower than most reported results.

I also find one bug. I think you are misleading by the Eq. (12)., the equation is used to compute each element of vector z.

```
kl_terms[t]=0.5*tf.reduce_sum(mu2+sigma2-2*logsigma,1)-T*.5 # each kl term is (1xminibatch)
```

should be

```
kl_terms[t]=0.5*tf.reduce_sum(mu2+sigma2-2*logsigma-1,1) # each kl term is (1xminibatch)
# kl_terms[t]=0.5*tf.reduce_sum(mu2+sigma2-2*logsigma,1)-T*z_size*.5 # alternatively
```

Or the kl term will blow up with large z_size.

Another issue is the mnist data in your code is not binarized. But it won't make much difference.

**Comment by lovecambi** [comment]
I know the reason. The sampled gaussian noise should be put into the function sampleQ.

**Comment by ericjang** [comment]
Thanks for the catch, sorry I didn't address this earlier. Would you mind
submitting a pull request to fix this?
